docs/integrations/gitlab/discovery.md

```diff
@@ -21,7 +21,8 @@ catalog:
     gitlab:
       yourProviderId:
         host: gitlab-host # Identifies one of the hosts set up in the integrations
-        branch: main # Optional. Uses `master` as default
+        branch: main # Optional. Uses `master` as default (fallback if default_branch is not set)
```

Comment:
It reads like `default_branch` is an option I can set in my config reading this but in reality it's returned from the GitLab API right? Perhaps we can rephrase this slightly

Reply:
Is `branch` always only used as fallback for the case if there is no `default_branch` configured for the repo at GitLab?

I struggle a bit with the naming here also considering `forceBranch`.

To me, it sounds like we should maybe call them `defaultBranch` and `branch` instead.

Also, considering your example using `latest` -- maybe this should be the default branch then? 🤔 Of course, depends on the setup used.

Esp. as `forceBranch` is a global option here. Of course, you could add multiple provider configs to have one where you want to use a non-default branch vs others where you might want to use the default branch / main branch.

```
# Optional. Sets forced branch to fetch project.
```
I would suggest to make this a bit clearer considering what I wrote above.

Just some thoughts.

Reply:
> Is branch always only used as fallback for the case if there is no default_branch configured for the repo at GitLab?
> 
> I struggle a bit with the naming here also considering forceBranch.
> 
> To me, it sounds like we should maybe call them defaultBranch and branch instead.


Yes, naming is strange, but it because of backward compatibility. [Please see this](https://github.com/backstage/backstage/issues/13587#issuecomment-1288980535)

> Esp. as forceBranch is a global option here. Of course, you could add multiple provider configs to have one where you want to use a non-default branch vs others where you might want to use the default branch / main branch.
>I would suggest to make this a bit clearer considering what I wrote above.

Good point

Reply:
Just want to point out again that `(fallback if default_branch is not set)` is not obvious what it means to someone not reading the code or knowing that this is something that the API returns. Need some extra clarification like.
> This name is used as fallback if the GitLab API does not return a default branch.